src/lib/zf/attr.c

```diff
@@ -281,14 +282,25 @@ int zf_attr_set_str(struct zf_attr* attr, const char* name, const char* val)
       return -ENOMSG;
     }
     char** p = get_field_str(attr, f);
-    free(*p);
     if( val != NULL ) {
+      if( strcmp(name, "log_file")==0 ) {
+        int rc = zf_log_redirect(val);
+        if( rc < 0 ) {
+          zf_log_stack_err(NO_STACK, "%s: Failed to redirect logging: %s\n",
+                          __func__, strerror(-rc));
+          /* Failure here is non-fatal. */
+          return rc;
+        }
+      }
+      free(*p);
```

Comment:
both if/else paths call free(*p); so think this can be moved straight after get_field_str as before.

Reply:
done. Moved to where it was before.